server/neptune/workflows/internal/terraform/workflow.go

```diff
@@ -236,14 +232,37 @@ func (r *Runner) Run(ctx workflow.Context) error {
 	}()
 
 	planResponse, err := r.Plan(ctx, root, response.ServerURL)
-
 	if err != nil {
-		return errors.Wrap(err, "running plan job")
+		return toExternalError(err, "running plan job")
 	}
 
 	if err := r.Apply(ctx, root, response.ServerURL, planResponse.PlanFile); err != nil {
-		return errors.Wrap(err, "running apply job")
+		return toExternalError(err, "running apply job")
 	}
 
 	return nil
 }
+
+// toExternalError allows callers of the workflow to handle specific error types
+// in whatever fashion.
+// we use errors.As here to ensure that we're accounting for wrapped errors
+func toExternalError(err error, msg string) error {
```

Comment:
nit: thoughts on using switch statement to check for the error type?

Reply:
This has changed btw

Reply:
Also not sure how i would even use a switch statement for this.